src/core/textures/BaseTexture.js

```diff
@@ -687,21 +692,24 @@ export default class BaseTexture extends EventEmitter
      * @static
      * @param {HTMLCanvasElement} canvas - The canvas element source of the texture
      * @param {number} scaleMode - See {@link PIXI.SCALE_MODES} for possible values
+     * @param {string} [origin='canvas'] - A string origin of who created the base texture
      * @return {PIXI.BaseTexture} The new base texture.
      */
-    static fromCanvas(canvas, scaleMode)
+    static fromCanvas(canvas, scaleMode, origin = 'canvas')
     {
+        const name = `${origin}_`;
+
         if (!canvas._pixiId)
         {
-            canvas._pixiId = `canvas_${uid()}`;
+            canvas._pixiId = `${name}${uid()}`;
```

Comment:
Looks like `name` is only used here, can we just remove this `name` and do `${origin}_${uid()}` here instead? That will reduce the string concats by one.